udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue

```diff
@@ -0,0 +1,213 @@
+          type="primary"
+          >
+            <p class="fr-mt-1v fr-mb-3v">
+              {{ t("Try to reset filters to widen your search.") }}<br/>
+              {{ t("You can also give us more details with our feedback form.") }}
+            </p>
+            <template v-slot:actions>
+              <button @click="resetFilters" class="fr-btn fr-btn--secondary">
+                {{ t("Reset filters") }}
+              </button>
+              <a :href="data_search_feedback_form_url" class="fr-btn fr-btn--tertiary-no-outline fr-btn--icon-left fr-icon-lightbulb-line fr-ml-1w">
+                {{ t("Tell us what you are looking for") }}
+              </a>
+            </template>
+          </ActionCard>
+        </div>
+      </section>
+    </div>
+  </form>
+</template>
+
+<script setup lang="ts">
+import { Dataservice, DataserviceCard, Pagination } from "@datagouv/components/ts";
+import { ref, onMounted, computed, useId, useTemplateRef, watchEffect, watch } from "vue";
+import { useI18n } from 'vue-i18n';
+import Loader from "../dataset/loader.vue";
+import MultiSelect from "../MultiSelect/MultiSelect.vue";
+import ActionCard from "../Form/ActionCard/ActionCard.vue";
+import { data_search_feedback_form_url } from "../../config";
+import { api } from "../../plugins/api";
+import franceWithMagnifyingGlassIcon from "../../../../templates/svg/illustrations/france_with_magnifying_glass.svg";
+import { PaginatedArray } from "../../api/types";
+import { refDebounced } from '@vueuse/core'
+
+const { t } = useI18n();
+
+const organization = ref<string | null>(null);
+const isRestricted = ref<boolean | null>(null);
+const isRestrictedId = useId();
+
+const page = ref(1);
+
+const hasFilters = computed(() => {
+  return organization.value || isRestricted.value !== null
+});
+const resetFilters = () => {
+  organization.value = '';
+  isRestricted.value = null;
+}
+
+const searchId = useId();
+const searchQuery = ref('');
+const searchQueryDebounced = refDebounced(searchQuery, 500);
+const searchInput = useTemplateRef('searchInput');
+
+const url = computed(() => {
+  const filters: { organization?: string, q?: string, is_restricted?: string, page?: string } = {}
+  if (organization.value) {
+    filters.organization = organization.value;
+  }
+  if (searchQueryDebounced.value) {
+    filters.q = searchQueryDebounced.value;
+  }
+  if (isRestricted.value !== null) {
+    filters.is_restricted = isRestricted.value.toString();
+  }
+  if (page.value && page.value !== 1) {
+    filters.page = page.value.toString();
+  }
+  const params = new URLSearchParams(filters);
+
+  let url = new URL(window.location.href);
+  url.search = params.toString();
+  window.history.pushState(null, "", url);
+
+  return `/dataservices?${params}`
+})
+
+const dataservices = ref<null | PaginatedArray<Dataservice>>(null);
+
+const search = async () => {
+  dataservices.value = null
+  const response = await api.get(url.value);
```

Comment:
You can add an [AbortController from axios](https://axios-http.com/docs/cancellation) to cancel unwanted request. The helper already exists in `udata_front/theme/gouvfr/datagouv-components/src/api/api.ts`.

Reply:
What is the point of the abort controller? Is it really useful in this case with a debounce of 500ms?

Reply:
Done :-)